openstack/image/v2/images/results.go

```diff
@@ -112,15 +120,48 @@ func (r *Image) UnmarshalJSON(b []byte) error {
 	}
 	*r = Image(s.tmp)
 
-	switch t := s.SizeBytes.(type) {
-	case nil:
-		r.SizeBytes = 0
-	case float32:
-		r.SizeBytes = int64(t)
-	case float64:
-		r.SizeBytes = int64(t)
-	default:
-		return fmt.Errorf("Unknown type for SizeBytes: %v (value: %v)", reflect.TypeOf(t), t)
+	// some of the values may be nil, or bool as a string format, or int as
+	// a string format so we need to assert them to the correct type using
+	// the assert package
+	r.Name, err = assert.String(s.Name, "Name")
+	if err != nil {
+		return err
+	}
```

Comment:
Is this necessary? What's the difference between this and regular json unmarshalling with `omitempty`?

Ditto all other uses of String below.